src/controller/java/src/chip/devicecontroller/model/ClusterState.java

```diff
@@ -38,6 +41,14 @@ public Map<Long, EventState> getEventStates() {
     return events;
   }
 
+  public void setDataVersion(int version) {
+    dataVersion = Optional.of(version);
+  }
+
+  public int getDataVersion() {
+    return dataVersion.get();
```

Comment:
what if dataVersion is missing? This would throw a `NoSuchElementException`. Should we use `orElese(0).get()` ? Or is 0 a valid data version.

Reply:
just return optional dataversion, application would do further check, 0 is valid version.